### DISCUSSION TITLE
Error running first artillery test

### DISCUSSION THREAD

**Discussion by MehseinJohan** [artilleryio/artillery#1884] [Artillery]
I've followed the installation guide and have seen the artillery dino command working correctly, but on running my first test I get a stacktrace indicating something is wrong.

I've checked my Node version (v18.14.1). Could you please help me ?
Many thanks in advance, I'm joining the files attached :  :

qa_automation_testing> npm install -D artillery@latest
npm WARN idealTree Removing dependencies.artillery in favor of devDependencies.artillery
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.

added 525 packages, removed 141 packages, changed 42 packages, and audited 720 packages in 45s

72 packages are looking for funding
  run `npm fund` for details

4 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
qa_automation_testing> artillery run browser-load-test.yml
Test run id: twa37_qj8c8c3ejznwawyet6fh8j68d6t8q_8fhz
- worker error, id: 1 Error: Cannot find module qa_automation_testing\flows.js'
Require stack:
-\AppData\Roaming\npm\node_modules\artillery\node_modules\@artilleryio\int-core\lib\runner.js
- \AppData\Roaming\npm\node_modules\artillery\lib\artillery-global.js
- \AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1047:15)
    at Module._load (node:internal/modules/cjs/loader:893:27)
    at Module.require (node:internal/modules/cjs/loader:1113:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at loadProcessor (AppData\Roaming\npm\node_modules\artillery\node_modules\@artilleryio\int-core\lib\runner.js:87:23)
    at prepare (AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js:109:18)
    at async MessagePort.onMessage (AppData\Roaming\npm\node_modules\artillery\lib\platform\local\worker.js:50:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'AppData\\Roaming\\npm\\node_modules\\artillery\\node_modules\\@artilleryio\\int-core\\lib\\runner.js',
    'AppData\\Roaming\\npm\\node_modules\\artillery\\node_modules\\@artilleryio\\int-core\\index.js',
    'AppData\\Roaming\\npm\\node_modules\\artillery\\lib\\artillery-global.js',
    'AppData\\Roaming\\npm\\node_modules\\artillery\\lib\\platform\\local\\worker.js'
  ]

[qa_automation_testing.zip](https://github.com/artilleryio/artillery/files/11080904/qa_automation_testing.zip)

**Comment by hassy** [comment]
hi @MehseinJohan, the custom JS code is in a file called `flow.js`, but the YAML file refers to `flows.js`. If you change it to `flow.js` it should work.

**Reply by MehseinJohan** [reply]
Hello @hassy , apologies for this mistake. I've re-run the script and I now have :

Test run id: tcayn_5jbkxef8nh768dq6aea7x8cx5wqby_kakt
Phase started: unnamed (index: 0, duration: 1s) 14:10:15(+0200)

Phase completed: unnamed (index: 0, duration: 1s) 14:10:16(+0200)

- TypeError: fn is not a function
    at Array.scenario (AppData\Roaming\npm\node_modules\artillery-engine-playwright\index.js:139:15)
--------------------------------------
--------------------------------------

vusers.created: ................................................................ 1
vusers.created_by_name.0: ...................................................... 1
vusers.failed: ................................................................. 1


Warning: multiple batches of metrics for period 1680005410000 2023-03-28T12:10:10.000Z
All VUs finished. Total time: 3 seconds

--------------------------------
Summary report @ 14:10:20(+0200)
--------------------------------

vusers.created: ................................................................ 1
vusers.created_by_name.0: ...................................................... 1
vusers.failed: ................................................................. 1

Could you please help ?

Thank you, 

Regards,

Johan

**Comment by MehseinJohan** [comment]
Hello team, 

It's now working but I do not have all the metrics (lcp, fcp...), I have only a few of them in fact :

--------------------------------
Summary report @ 17:53:47(+0200)
--------------------------------

browser.http_requests: ......................................................... 169
vusers.completed: .............................................................. 10
vusers.created: ................................................................ 10
vusers.created_by_name.Dev account signup: ..................................... 10
vusers.failed: ................................................................. 0
vusers.session_length:
  min: ......................................................................... 2909.3
  max: ......................................................................... 4428.3
  median: ...................................................................... 3011.6
  p95: ......................................................................... 3605.5
  p99: ......................................................................... 3605.5

Is there a reason ?

Thank you in advance,

Johan

**Reply by hassy** [reply]
that's odd! there should be front-end metrics in there as `vusers.failed` is `0`. I'll try running the tests to see what might be going wrong.
